src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -1304,6 +1327,14 @@ static ExtrusionEntityCollection traverse_extrusions(const PerimeterGenerator& p
                 }
                 assert(extrusion_loop.paths.front().first_point() == extrusion_loop.paths.back().last_point());
 
+				//This is for staggered layers. 
+				//All odd perimeters are staggerd up by half the layer height                
+                if (extrusion->inset_idx % 2 == 1 && perimeter_generator.config->staggered_perimeters) {
```

Comment:
Would be better to do `perimeter_generator.config->staggered_perimeters` check first, the it won't waste computation on odd check when the option is off.